src/Yardarm/Enrichment/Schema/Internal/BaseTypeEnricher.cs

```diff
@@ -28,6 +28,13 @@ public BaseTypeEnricher(GenerationContext context)
 
             BaseTypeSyntax[] additionalBaseTypes = feature.GetBaseTypes(context.LocatedElement).ToArray();
 
+            if (target.BaseList != null)
+            {
+                additionalBaseTypes = additionalBaseTypes.Where(additionalBaseType =>
+                    !target.BaseList.Types.Any(currentType => additionalBaseType.ToString() == currentType.ToString()))
```

Comment:
Is a string comparison the best option we have here? I feel like there should be a cleaner way, this might not account well for namespaces.

Also, I'd drop the ToArray above and make additionalBaseTypes an IEnumerable. Then we don't need to allocate a second array here, we can just do it below in the call to AddBaseListTypes.

Reply:
Updated per your suggestions